runtime/lua/vim/lsp/codelens.lua

```diff
@@ -31,10 +31,24 @@ local function execute_lens(lens, bufnr, client_id)
   local line = lens.range.start.line
   api.nvim_buf_clear_namespace(bufnr, namespaces[client_id], line, line + 1)
 
+  local command = lens.command
+  local fn = vim.lsp.commands[command.command]
+  if fn then
+    fn(command, { bufnr = bufnr, client_id = client_id })
+    return
+  end
   -- Need to use the client that returned the lens → must not use buf_request
   local client = vim.lsp.get_client_by_id(client_id)
   assert(client, 'Client is required to execute lens, client_id=' .. client_id)
-  client.request('workspace/executeCommand', lens.command, function(...)
+  local command_provider = client.server_capabilities.executeCommandProvider
+  local commands = type(command_provider) == 'table' and command_provider.commands or {}
+  if not vim.tbl_contains(commands, command.command) then
+    vim.notify(string.format(
+      "Language server doesn't support command `%s`. This command may require a client extension", command.command),
```

Comment:
how do we feel about contractions? Also missing a terminating period.

Reply:
Not sure? Happy to rephrase it if you've a suggestion.

Reply:
I was only suggesting "does not" and adding a period at the end

Reply:
Oh gotcha. Changed it.